aten/src/ATen/native/nested/NestedTensorMath.h

```diff
@@ -32,6 +34,37 @@ inline at::Tensor get_buffer(const at::Tensor& tensor) {
   return get_nested_tensor_impl(tensor)->get_buffer();
 }
 
+  /**
+   * Create a new nested tensor that is a view of a base nested tensor
+   *
+   * create_view_tensor calls a specialized constructor that copys the
+   * the keys from base onto the new view tensor being created.
+   * The storage is shared between the base and the returned view tensor
+   *
+   * All callers of this helper must:
+   * - Only return a view of the input
+   * - Must be explicit and define a derivative
+   *
+   * @param base Base tensor to construct view from.
+   * @param nested_size_tensor View tensors' sizes.
+   * @param nested_stride_tensor View tensors' strides.
+   * @param offsets View tensors' offsets.
+   * @return A newly constructed view tensor
+   */
+  inline at::Tensor create_nested_view_tensor(
+      const at::Tensor& base,
+      at::Tensor nested_size_tensor,
+      at::Tensor nested_stride_tensor,
+      std::vector<int64_t>&& offsets) {
+    TORCH_INTERNAL_ASSERT(base.is_nested(), "This function can only be used to create nested tensor views");
```

Comment:
nit: we might want to add an assert that the autograd key is in the exclude set here. That will ensure that we never use this wrong.

Reply:
But isn't that not true. the concrete use of this is here: 
https://github.com/drisspg/pytorch/blob/e71e32966a79aab2b12a223fb9bd4a59c4aadf22/aten/src/ATen/native/nested/NestedTensorMath.cpp#L1004

We need the autograd keys to get carried through. Unless you mean Autograd(Backend) and are not talking about AutogradNestedTensor?

Reply:
The autograd layer adds the autograd key to the exclude set when you go through it. So anything that is not `CompositImplicit` will have autograd in the exclude set.

Reply:
Gotcha, added.